### ISSUE TITLE
C++ Code Highlighting Fails For Large Files - Broken Again

### ISSUE THREAD

**Issue by ArashPartow** [Microsoft/vscode#21443]
In regards to issue #11597 

I updated VSCode to 1.91, and C/C++ for Visual Studio Code  0.10.1.

When opening large files. The start (roughly first 10k lines) looks ok, but towards the end everything starts to go BLUE. 

This image is at the start of the file: http://pasteboard.co/CXVwq2zzU.png

This image is at the end of the file: http://pasteboard.co/CXW1KVxHg.png

Notice the difference in colouring of brackets ( {}, [] and ()) and the excessive blue in pretty much every other token. It's as if the editor has gone to Smurfsville post line 10k. 

The files used:
- https://raw.githubusercontent.com/ArashPartow/exprtk/master/exprtk.hpp
- https://github.com/ArashPartow/strtk/blob/master/strtk.hpp
- https://github.com/ArashPartow/wykobi/blob/master/wykobi.inl

In short: Would it be *possible* to do whatever it is you're doing for the first ~10K lines for all the lines in the file?  

As I don't believe there's any need for differential treatment of lines post 10K.

@ramya-rao-a 
@aeschli 
@alexandrudima 
@chrmarti 
@joaomoreno

**Comment by aeschli** [comment]
The problem is a duplicate of https://github.com/atom/language-c/issues/185
